src/tycho/apps/ingestion/application/usecases/load_documents.py

```diff
@@ -0,0 +1,31 @@
+"""LoadDocuments usecase."""
+
+from apps.ingestion.containers import IngestionContainer
+from core.entities.document import DocumentType
+from core.interfaces.document_repository_interface import IUpsertResult
+
+
+class LoadDocumentsUsecase:
+    """Usecase for loading and persisting documents."""
+
+    def __init__(self, container: IngestionContainer):
+        """Initialize with container."""
+        self.container = container
+
+    def execute(self, document_type: DocumentType) -> IUpsertResult:
+        """Execute the usecase to load and persist documents."""
+        try:
+            repository = self.container.document_repository()
+
+            documents = repository.fetch_by_type(document_type)
+
+            result = repository.upsert_batch(documents)
+
+            return result
```

Comment:
Generally speaking try/except blocks should contain a single statement, and eventually multiple exceptions. By doing so, exceptions are more relevant and scoped.